modules/core/server/controllers/mongoose-filtering.server.controller.js

```diff
@@ -0,0 +1,54 @@
+﻿// Service for managing Filtering/Parameterized-Queries.
+
+var _query = null;
+var _modifiedQuery = null;
+var _totalDocumentCount = null;
+var _page = 0;
+var _take = 50; // default to prevent unreasonable take() requests
```

Comment:
Any client request can set the *take*, but I think the server should have the responsibility of managing a maximum *take*. This would protect from malicious, or erroneous, requests.

I've set the maximum in this case to be 150. However, I think it should be higher. Either way, it should be configurable to each implementation of the `mongoose-filtering` functionality; i.e. the calling controllers should be in control of the maximum & default *take*.